data/state/accountsDB.go

```diff
@@ -112,7 +112,18 @@ func (adb *AccountsDB) LoadDataTrie(accountHandler AccountHandler) error {
 		return NewErrorTrieNotNormalized(HashLength, len(accountHandler.GetRootHash()))
 	}
 
-	dataTrie, err := adb.mainTrie.Recreate(accountHandler.GetRootHash())
+	dataTrie, err := adb.searchDataTrieInJournal(accountHandler.GetRootHash())
+	if err == nil {
+		accountHandler.SetDataTrie(dataTrie)
+		return nil
+	}
+
+	_, isErrMissingTrie := err.(*ErrMissingTrie)
```

Comment:
can you change these tests ? i do not like handling different type of errors in different ways.

Reply:
I follow the rule: output only (pointer, nil) or (nil, error). So, even though they are not pretty, they conform to how our program has been written. (In other cases I have output an "empty" kind of value for pointer but here should not be applied)